lib/utils/resolveSilent.cjs

```diff
@@ -0,0 +1,50 @@
+// NOTICE: This file is generated by Rollup. To modify it,
+// please instead edit the ESM counterpart and rebuild with Rollup (npm run build).
+'use strict';
+
+const fs = require('node:fs');
+const path = require('node:path');
+const node_url = require('node:url');
+const importMetaResolve = require('@dual-bundle/import-meta-resolve');
+const resolveFrom = require('resolve-from');
+
+const pathSuffixes = ['', '.js', '.json', `${path.sep}index.js`, `${path.sep}index.json`];
+
+const specifierSuffixes = ['', '.js', '.json', '/index.js', '/index.json'];
+
+/**
+ * @param {string} parent
+ * @param {string} lookup
+ * @return {string | undefined}
+ */
+function resolveSilent(parent, lookup) {
+	if (path.isAbsolute(lookup)) {
+		for (const suffix of pathSuffixes) {
+			const filename = lookup + suffix;
+
+			if (fs.existsSync(filename)) {
+				return filename;
+			}
+		}
+
+		return;
+	}
+
+	const resolvedParent = node_url.pathToFileURL(path.resolve(parent, 'noop.js'));
+
+	for (const suffix of specifierSuffixes) {
+		try {
+			return node_url.fileURLToPath(importMetaResolve.moduleResolve(lookup + suffix, resolvedParent));
+		} catch {
+			//
+		}
+	}
+
+	/**
+	 * Yarn P'n'P does not support pure ESM well by default, this is only a workaround for it
+	 * @see https://github.com/wooorm/import-meta-resolve/issues/23
+	 */
+	return resolveFrom.silent(parent, lookup);
```

Comment:
[ask] This workaround will no longer be needed when Stylelint becomes pure ESM, right?

Reply:
Nope. `import-meta-resolve` does not support such options.

https://github.com/wooorm/import-meta-resolve#differences-to-node

<img width="873" alt="image" src="https://github.com/stylelint/stylelint/assets/8336744/4a294582-8ddf-4939-91e3-7bd34495ceaa">

Reply:
Sure! How about `TODO: this workaround is still necessary util native `import.meta.resolve` available`

Reply:
Sounds good 👍🏼